### PR TITLE
feat: enable code cache for custom protocols

### PR DESCRIPTION
**Pull request by zcbenz**
#### Description of Change

Add an option in `protocol.registerSchemesAsPrivileged` to allow V8 code cache in custom schemes.

Upstream Chromium issue: https://bugs.chromium.org/p/chromium/issues/detail?id=1500437.

/cc @deepak1556 and @bpasero since there will need changes in VS Code.

Fixes https://github.com/electron/electron/issues/27075

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Add an option in `protocol.registerSchemesAsPrivileged` to allow V8 code cache in custom schemes.

### CODE REVIEW

**Comment by deepak1556** [comment]
Should we enforce a good default be only allowing code cache for standard custom schemes, since the cache key relies on origin lock of the process when available. Custom protocols with null origin will have their code cache grouped together and I am not sure cache will be useful in that case since there is more chance of collision.

**Comment by zcbenz** [comment]
> Should we enforce a good default be only allowing code cache for standard custom schemes, since the cache key relies on origin lock of the process when available.

I updated the API to throw when trying to enabling code cache for non-standard scheme.

**Review by deepak1556 — APPROVED** [review]

**Review by miniak — APPROVED** [review]

File: shell/browser/api/electron_api_protocol.cc [hunk 1]
```diff
@@ -104,6 +113,15 @@ void RegisterSchemesAsPrivileged(gin_helper::ErrorThrower thrower,
     return;
   }
 
+  for (const auto& custom_scheme : custom_schemes) {
+    if (custom_scheme.options.codeCache && !custom_scheme.options.standard) {
+      thrower.ThrowError(
+          "Code cache can noly be enabled when the custom "
```
Comment (anonrig) [inline shell/browser/api/electron_api_protocol.cc:119]:
```suggestion
          "Code cache can only be enabled when the custom "
```

**Review by jkleinsc — COMMENTED** [review]
API LGTM

**Review by samuelmaddock — APPROVED** [review]
API LGTM

**Pull request merged**